src/ClassApplication/Controllers/ClassController.cs

```diff
@@ -1,121 +1,260 @@
+        /// <param name="classroomName">Name of the classroom being created</param>
+        /// <param name="description">String to set the new classroom description to (Optional)</param>
+        [HttpPost("CreateClass")]
+        public async Task<ActionResult<string>> CreateClassroom(string ownerId, string classroomName, string description = "")
+        {
+            string classId = Guid.NewGuid().ToString();
+            Classroom classroom = new Classroom(classId, ownerId, classroomName, description);
+            classroom.AddStudent(ownerId);
+
+            await classesContainer.CreateItemAsync<Classroom>(classroom, new PartitionKey(classId));
+
+            return Ok();
+        }
+
+
+
+        /// <summary>
+        /// 
+        /// Returns data from a classroom from a given classroom ID
+        ///
+        /// </summary>
+        /// <param name="classId">classroom id</param>
+        /// <returns>An array containing a value determined by the parameter</returns>
+        [HttpGet("GetClassByID/{classId}")]
+        public async Task<Classroom> GetClassroom(string classId)
+        {
+            Classroom classroom;
+            classroom = await classesContainer.ReadItemAsync<Classroom>(classId, new PartitionKey(classId));
+
+            return classroom;
+        }
+
+        /// <summary>
+        /// 
+        /// Finds a classroom by the classrooms name
+        ///
+        /// </summary>
+        /// <param name="className">classroom name</param>
+        /// <returns>A list containing a classIDs determined by the parameter</returns>
+        [HttpGet("QueryClassByName/{className}")]
+        public async Task<List<string>> QueryClassByName(string className)
+        {
+            string sqlQueryStatement = "SELECT * FROM c WHERE c.Name = @className";
+            QueryDefinition query = new QueryDefinition(sqlQueryStatement).WithParameter("@className", className);
+            FeedIterator<Classroom> queryIterator = classesContainer.GetItemQueryIterator<Classroom>(query);
+
+            List<string> result = new List<string>();
+
+            while (queryIterator.HasMoreResults)
+            {
+                FeedResponse<Classroom> resultSet = await queryIterator.ReadNextAsync();
+
+                foreach (Classroom classroom in resultSet)
+                {
+                    result.Add(classroom.id);
+                }
+
+            }
+
+            return result;
+
+        }
+
+
+
+        /// <summary>
+        /// This call is used to add a Note to the Classroom
+        /// using the Note ID generated by the Notes Applicaton
+        /// </summary>
+        /// <param name="classId">Id of the classroom</param>
+        /// <param name="notesId">Id of the note being added</param>
+        [HttpPut("AddNoteToClass/{classId}/notes/{notesId}")]
+        public async Task<ActionResult<string>> AddNoteToClass(string classId, string notesId)
+        {
+
```

Comment:
Good job making this simple. When y’all start worrying about performance, I would like y’all to think about what the user of this API would have to do if they wanted to add multiple notes to a classroom at a time. Will they ever? Who knows. If they did though, how would they? They would have to make a bunch of HTTP calls. Do you think this is performant? Something to think about. Good work altogether though!